### PR TITLE
Fixed #872 : Support View document type

### LINKED ISSUE #872: View doesn't support documentType
View needs to support `View.documentType`. iOS and .NET have implemented the feature.

https://github.com/couchbase/couchbase-lite-java-core/blob/master/src/main/java/com/couchbase/lite/View.java#L105-L107

### PR DESCRIPTION
**Pull request by pasin**
Implement View.documentType for filtering by the document type when indexing the view.

#872

### CODE REVIEW

File: src/main/java/com/couchbase/lite/store/SQLiteViewStore.java [hunk 1]
```diff
@@ -534,17 +534,17 @@ public void emit(Object key, Object value) {
...
-                    try {
-                        emitBlock.setSequence(sequence);
-                        mapBlocks.get(i).map(curDoc, emitBlock);
-                    } catch (Throwable e) {
-                        String msg = String.format("Error when calling map block of view '%s'",
-                                view.getName());
-                        Log.e(Log.TAG_VIEW, msg, e);
-                        throw new CouchbaseLiteException(msg, e, new Status(Status.CALLBACK_ERROR));
+                        Log.v(Log.TAG_VIEW, "#%s: map '%s' for view %s...",
+                                sequence, docID, view.getName());
```
Comment (hideki) [inline src/main/java/com/couchbase/lite/store/SQLiteViewStore.java]: I don't recommend to rely on auto converting from long to String.

Comment (pasin) [inline src/main/java/com/couchbase/lite/store/SQLiteViewStore.java]: I have thought I converted them all. Fixing it now.

**Comment by pasin** [comment]
@hideki I have fixed the formatting per your comment. Please review. Thanks!

**Comment by pasin** [comment]
Reference: https://github.com/couchbase/couchbase-lite-java-core/issues/875

**Pull request merged**